open_bus_stride_api/routers/route_timetable.py

```diff
@@ -0,0 +1,74 @@
+import typing
+import datetime
+
+import pydantic
+import sqlalchemy
+from fastapi import APIRouter
+
+from open_bus_stride_db.model.gtfs_stop import GtfsStop
+from open_bus_stride_db import model
+
+from . import common
+
+
+router = APIRouter()
+
+
+class RouteTimetablePydanticModel(pydantic.BaseModel):
+    id: int
+    name: str
+    city: str
+    lon: int
+    lat: int
+    planned_arrival_time: datetime.datetime = None
+    gtfs_line_ref: str = None
+    gtfs_line_start_time: datetime.datetime = None
+    gtfs_ride_id: str = None
+
+LIST_MAX_LIMIT = 100
+WHAT_PLURAL = """the stops timetable of a given bus.
+Currently, only planned time (gtfs) is returned for every stop"""
+TAG = 'user cases'
+PYDANTIC_MODEL = RouteTimetablePydanticModel
+
+def _post_session_query_hook(session_query: sqlalchemy.orm.Query):
+    return (
+        session_query
+        .select_from(model.GtfsStop)
+        .add_entity(model.GtfsRideStop)
+        .add_entity(model.GtfsRide)
+        .add_entity(model.GtfsRoute)
+        .join(model.GtfsRideStop, model.GtfsRideStop.gtfs_stop_id == model.GtfsStop.id)
+        .join(model.GtfsRide, model.GtfsRide.id == model.GtfsRideStop.gtfs_ride_id)
+        .join(model.GtfsRoute, model.GtfsRoute.id == model.GtfsRide.gtfs_route_id)
+        .order_by((sqlalchemy.asc)(model.GtfsRideStop.arrival_time))
```

Comment:
we do ordering in [here](https://github.com/hasadna/open-bus-stride-api/blob/d0426d585a3f78da26bcb8ebcdd1fd4e8e4304f1/open_bus_stride_api/routers/common.py#L56), as part of the get_list functionality - this is because to support pagination we must have a defined order for every query. So you should not do ordering here, but instead use the mechanism in get_list

Reply:
Looks like arrival_time field is not indexed, this results in very long query time when a large number of results is returned (for example, when running this method without any filters). Please add an index on it

Reply:
The reason I didn't use the regular order_by, is because it only supports ordering by the "origin" db model (gtfs stop).
I  initially thought about adding a parsing for the joined tables, to allow strings like "gtfs_ride_stop.arrival_time asc", but this is a bit weird and requires too much familiarity with the tables structure (which is exactly what I don't want in those "user cases" apis).

Reply:
As far as I see, we always add an "id desc" ordering at the end if it doesn't yet exist, and this logic comes after the post_session_query_hook I added, so the order should be consistent.
Maybe to be more explicit, instead of pushing the order by func under post_session_query_hook,  I can either-
*  Split the order by to "order_by_string" (current) and "order_by_funcs" and then make sure I only use one of them. 
* Break the current API, and change order_by to receive a func instead of a string, and the string parsing would be done externally.
What do you think?

Reply:
I think you can add an argument to get_list: `skip_order_by` - and if it's True it will skip all order_by logic